src/swap/SwapPending.tsx

```diff
@@ -0,0 +1,203 @@
+            >
+              {t('swapCompleteScreen.swapErrorModal.body')}
+            </Dialog>
+          </>
+        )
+      case SwapState.PRICE_CHANGE:
+        return (
+          <Dialog
+            isVisible={true}
+            title={t('swapCompleteScreen.swapPriceModal.title')}
+            actionText={t('swapCompleteScreen.swapPriceModal.action')}
+            actionPress={navigateToReviewScreen}
+            testID="PriceChangeModal"
+            onBackgroundPress={navigateToReviewScreen}
+          >
+            {t('swapCompleteScreen.swapPriceModal.body')}
+          </Dialog>
+        )
+      // These states are the first and last of the swap flow
+      // Avoids a flicker when resetting state.swap.swapInfo
+      case SwapState.USER_INPUT:
+      case SwapState.COMPLETE:
+        return (
+          <>
+            <Text style={styles.text}>{t('swapCompleteScreen.swapSuccess')}</Text>
+            <Button
+              type={BtnTypes.SECONDARY}
+              text={t('swapCompleteScreen.swapAgain')}
+              onPress={navigateToSwapStart}
+            />
+          </>
+        )
+    }
+  }, [swapState])
+
+  const SwapIcon = useMemo(() => {
+    switch (swapState) {
+      default:
+      case SwapState.USER_INPUT:
+      case SwapState.START:
+      case SwapState.APPROVE:
+      case SwapState.EXECUTE:
+        return (
+          <ActivityIndicator
+            size="large"
+            color={colors.greenBrand}
+            testID="SwapPending/loadingIcon"
+            style={styles.activityIndicator}
+          />
+        )
+      case SwapState.ERROR:
+      case SwapState.PRICE_CHANGE:
+        return (
+          <View
+            testID="SwapPending/errorIcon"
+            style={[styles.iconContainer, { backgroundColor: colors.warning }]}
+          >
+            <Times color={colors.light} />
+          </View>
+        )
+      case SwapState.COMPLETE:
+        return (
+          <View
+            testID="SwapPending/completeIcon"
+            style={[styles.iconContainer, { backgroundColor: colors.greenUI }]}
+          >
+            <Checkmark color={colors.light} />
+          </View>
+        )
+    }
+  }, [swapState])
+
+  // Prevent back button on Android
+  useEffect(() => {
+    const backPressListener = () => true
+    BackHandler.addEventListener('hardwareBackPress', backPressListener)
+    return () => BackHandler.removeEventListener('hardwareBackPress', backPressListener)
+  }, [])
+
+  return (
+    <SafeAreaView style={styles.safeAreaView}>
+      <View style={styles.contentContainer}>
+        {SwapIcon}
+        {SwapDisplay}
+      </View>
+    </SafeAreaView>
+  )
+}
+
+const styles = StyleSheet.create({
+  safeAreaView: {
+    flex: 1,
+  },
+  contentContainer: {
+    alignItems: 'center',
+    justifyContent: 'center',
+    padding: Spacing.Regular16,
+    flexGrow: 1,
+  },
+  activityIndicator: {
+    left: 0,
+    right: 0,
+    top: 0,
+    bottom: 0,
```

Comment:
do we need this left/right/top/bottom 0 style when the position is not absolute? seems to look fine without these styles for me

Reply:
Fixed in cbf62316a150a46b33306cb983a3bd75c031b07b!